standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/impl/FindPotentialCompactionsFunction.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hive.metastore.txn.impl;
+
+import org.apache.hadoop.hive.metastore.api.MetaException;
+import org.apache.hadoop.hive.metastore.txn.CompactionInfo;
+import org.apache.hadoop.hive.metastore.txn.jdbc.MultiDataSourceJdbcResource;
+import org.apache.hadoop.hive.metastore.txn.jdbc.TransactionalFunction;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class FindPotentialCompactionsFunction implements TransactionalFunction<Set<CompactionInfo>> {
+  
+  private final int fetchSize;
+  private final int abortedThreshold;
+  private final long abortedTimeThreshold;
+  private final long lastChecked;
+  private final boolean collectAbortedTxns;
+
+  public FindPotentialCompactionsFunction(int fetchSize, int abortedThreshold, long abortedTimeThreshold, long lastChecked, boolean collectAbortedTxns) {
+    this.fetchSize = fetchSize;
```

Comment:
this is a function, wouldn't it be better if we structure it as a Builder
````
new FindPotentialCompactionsFunction()
  .withLastChecked(lastChecked)
  ......
  .withFetchSize(fetchSize)
  .execute(jdbcResource);
````

Reply:
that is minor, just to consider as list of params in constructor is long

Reply:
Shortened the length or arguments, I think now it should be fine without the builder pattern.